File: apps/simulation/internal/model/control.go
package model

import "time"

const SimulationSafetyDisclaimer = "Simulation-only interface. No real plant control."

type ControlMode string

const (
	ControlModeManual   ControlMode = "MANUAL"
	ControlModeAuto     ControlMode = "AUTO"
	ControlModeDisabled ControlMode = "DISABLED"
)

type ControlAuthority string

const (
	ControlAuthorityUser     ControlAuthority = "USER"
	ControlAuthorityScenario ControlAuthority = "SCENARIO"
	ControlAuthorityPID      ControlAuthority = "PID"
	ControlAuthoritySystem   ControlAuthority = "SYSTEM"
	ControlAuthorityNone     ControlAuthority = "NONE"
)

type CommandRejectReason string

const (
	CommandRejectControlModeAuto       CommandRejectReason = "CONTROL_MODE_AUTO"
	CommandRejectControlDisabled       CommandRejectReason = "CONTROL_DISABLED"
	CommandRejectUnsupportedSource     CommandRejectReason = "UNSUPPORTED_COMMAND_SOURCE"
	CommandRejectUnsupportedTarget     CommandRejectReason = "UNSUPPORTED_TARGET"
	CommandRejectInvalidCommand        CommandRejectReason = "INVALID_COMMAND"
	CommandRejectInvalidPayload        CommandRejectReason = "INVALID_PAYLOAD"
	CommandRejectTargetControlledByPID CommandRejectReason = "TARGET_CONTROLLED_BY_PID"
	CommandRejectUnknown               CommandRejectReason = "UNKNOWN"
)

type ControlStatus struct {
	ControllerTag          string           `json:"controllerTag"`
	ControlledVariableTag  string           `json:"controlledVariableTag"`
	ManipulatedVariableTag string           `json:"manipulatedVariableTag"`
	Mode                   ControlMode      `json:"mode"`
	Authority              ControlAuthority `json:"authority"`
	Enabled                bool             `json:"enabled"`
	PIDImplemented         bool             `json:"pidImplemented"`
	Reason                 string           `json:"reason"`
	UpdatedAt              time.Time        `json:"updatedAt"`
	UpdatedBy              string           `json:"updatedBy"`
	SafetyDisclaimer       string           `json:"safetyDisclaimer"`
}

type ModeChangeRequest struct {
	Mode        ControlMode `json:"mode"`
	RequestedBy string      `json:"requestedBy,omitempty"`
	Reason      string      `json:"reason,omitempty"`
}

type ArbitrationDecision struct {
	Allowed     bool                `json:"allowed"`
	Reason      CommandRejectReason `json:"reason,omitempty"`
	Message     string              `json:"message,omitempty"`
	Mode        ControlMode         `json:"mode"`
	Authority   ControlAuthority    `json:"authority"`
	TargetTag   string              `json:"targetTag"`
	CommandType CommandType         `json:"commandType"`
	Source      CommandSource       `json:"source"`
}

File: apps/web/src/test/fixtures/control.ts
import type { ControlStatus } from "@/entities/control/model/types";

export const manualControlStatusFixture: ControlStatus = {
  controllerTag: "TIC-101",
  controlledVariableTag: "TT-101",
  manipulatedVariableTag: "V-101.POS",
  mode: "MANUAL",
  authority: "USER",
  enabled: true,
  pidImplemented: true,
  reason: "Operator manual control",
  updatedAt: "2026-05-21T06:00:00Z",
  updatedBy: "e2e-test",
  safetyDisclaimer: "Simulation-only. No real plant control.",
};

export const autoControlStatusFixture: ControlStatus = {
  ...manualControlStatusFixture,
  mode: "AUTO",
  authority: "PID",
  reason: "Simulation-only PID authority",
};

export const disabledControlStatusFixture: ControlStatus = {
  ...manualControlStatusFixture,
  mode: "DISABLED",
  authority: "NONE",
  enabled: false,
  reason: "Control output disabled in simulation",
};
